user_connections.go

```diff
@@ -0,0 +1,161 @@
+package sockchat
+
+import (
+	"log"
+	"sync"
+)
+
+// SockchatUserHandler manages user actions
+type SockchatUserHandler interface {
+	MakeRequest(action string, payload any) error
+	Write(msg SocketMessage)
+	AddConnection(conn *SockChatWS)
+	RemoveConnection(conn *SockChatWS)
+	getActiveConnectionsCount() int
+	getNick() string
+}
+
+// ConnectedUsersPool is responsible for tracking all user handlers
+type ConnectedUsersPool struct {
+	handlers map[string]SockchatUserHandler
+	lock     sync.RWMutex
+}
+
+func NewUserManager() *ConnectedUsersPool {
```

Comment:
```suggestion
func NewConnectedUsersPool() *ConnectedUsersPool {
```
This constructor name is misleading